auxlib/packaging.py

```diff
@@ -1,101 +1,142 @@
-#             subprocess.check_call(['git', 'diff', '--cached', '--quiet'])
-#             return False
-#         except subprocess.CalledProcessError:
-#             return True
-
-#     def _get_most_recent_tag():
-#         try:
-#             return subprocess.check_output(["git", "describe", "--tags"]).strip()
-#         except subprocess.CalledProcessError as e:
-#             if e.returncode == 128:
-#                 return "0.0.0.0"
-#             else:
-#                 raise
-
-#     def _get_hash():
-#         try:
-#             return subprocess.check_output(["git", "rev-parse", "HEAD"]).strip()[:7]
-#         except subprocess.CalledProcessError:
-#             return
-
-#     tag = _get_most_recent_tag()
-#     m = re.match("(?P<xyz>\d+\.\d+\.\d+)(?:-(?P<dev>\d+)-(?P<hash>.+))?", tag)
-
-#     version = m.group('xyz')
-#     if m.group('dev') or _is_dirty():
-#         version += ".dev{dev}+{hash}".format(dev=m.group('dev') or 0,
-#                                              hash=m.group('hash') or _get_hash())
-
-#     return version
-
-
-# def is_git_repo(path):
-#     if path == '/':
-#         return False
-#     else:
-#         return os.path.isdir(os.path.join(path, '.git')) or is_git_repo(os.path.dirname(path))
-
-
-# def get_version(package_name):
-#     """Returns a version string for the current package, derived
-#     either from the SCM (git currently) or from PKG-INFO.
-
-#     This function is expected to run in two contexts. In a development
-#     context, where .git exists, the version is pulled from git tags
-#     and written into PKG-INFO to create an sdist or bdist.
-
-#     In an installation context, the PKG-INFO file written above is the
-#     source of version string.
-
-#     """
-
-#     here = os.path.abspath(os.path.expanduser(os.path.dirname(__file__)))
-#     if is_git_repo(here):
-#         return _get_version_from_git_tag()
-
-#     version_from_pkg = _get_version_from_pkg_info(package_name)
-#     if version_from_pkg:
-#         return version_from_pkg
-
-#     raise RuntimeError("Could not get package version (no .git or .version file)")
-
-
-# def strip_comments(l):
-#     return l.split('#', 1)[0].strip()
-
-
-# def requirements(*f):
-#     return [
-#         r for r in (
-#             strip_comments(l) for l in open(
-#                 os.path.join(os.getcwd(), 'requirements', *f)).readlines()
-#         ) if r]
-
-
-# # if __name__ == "__main__":
-# #     print(get_version())
+# -*- coding: utf-8 -*-
+from __future__ import print_function, division, absolute_import
+from logging import getLogger
+from os.path import basename, dirname, join, isdir
+from pkg_resources import resource_string
+from re import match
+from setuptools.command.build_py import build_py
+from setuptools.command.sdist import sdist
+from setuptools.command.test import test as TestCommand
+from subprocess import CalledProcessError, check_call, check_output
+from sys import exit
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Redefining built-in 'exit'](https://www.codacy.com/app/kalefranz/auxlib/file/2131389828/issues/source?bid=3138661&fileBranchId=3139130#l11)